FreeRTOS/Test/CMock/smp/task_creation_covg/task_creation_covg_utest.c

```diff
@@ -0,0 +1,335 @@
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+/*! @file task_creation_covg_utest */
+
+/* C runtime includes. */
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+/* Task includes */
+#include "FreeRTOS.h"
+#include "FreeRTOSConfig.h"
+#include "event_groups.h"
+#include "queue.h"
+//#include "portasm.h"
+
+/* Test includes. */
+#include "unity.h"
+#include "unity_memory.h"
+#include "../global_vars.h"
+#include "../smp_utest_common.h"
+
+/* Mock includes. */
+#include "mock_timers.h"
+#include "mock_fake_assert.h"
+#include "mock_fake_port.h"
+
+
+/* ===========================  EXTERN VARIABLES  =========================== */
+extern volatile UBaseType_t uxDeletedTasksWaitingCleanUp;
+extern volatile UBaseType_t uxSchedulerSuspended;
+extern volatile TCB_t *  pxCurrentTCBs[ configNUMBER_OF_CORES ];
+extern volatile BaseType_t xSchedulerRunning;
+extern volatile TickType_t xTickCount;
+extern volatile UBaseType_t uxCurrentNumberOfTasks;
+
+/* ==============================  Global VARIABLES ============================== */
+TaskHandle_t xTaskHandles[configNUMBER_OF_CORES] = { NULL };
+
+/* ============================  Unity Fixtures  ============================ */
+/*! called before each testcase */
+void setUp( void )
+{
+    commonSetUp();
+}
+
+/*! called after each testcase */
+void tearDown( void )
+{
+    commonTearDown();
+}
+
+/*! called at the beginning of the whole suite */
+void suiteSetUp()
+{
+}
+
+/*! called at the end of the whole suite */
+int suiteTearDown( int numFailures )
+{
+    return numFailures;
+}
+
+/* ===========================  EXTERN FUNCTIONS  =========================== */
+extern void vTaskEnterCritical(void);
+
+/* ==============================  Helper functions for Test Cases  ============================== */
+
+
+/* ==============================  Test Cases  ============================== */
+
+/*
+The kernel will be configured as follows:
+    #define configNUMBER_OF_CORES                           (N > 1)
+    #define configUSE_CORE_AFFINITY                         1
+    #define configUSE_TASK_PREEMPTION_DISABLE               1
+
+Coverage for
+    static void prvAddNewTaskToReadyList( TCB_t * pxNewTCB )
+    covers the case where the task being created is not the first or only task.
+*/
+void test_create_two_tasks_with_the_first_suspended( void )
+{
+    TaskHandle_t xTaskHandles[configNUMBER_OF_CORES] = { NULL };
+
+    xTaskCreate( vSmpTestTask, "SMP Task", configMINIMAL_STACK_SIZE, NULL, 1, &xTaskHandles[0] );
+    vTaskSuspend(xTaskHandles[0]);
+
+    xTaskCreate( vSmpTestTask, "SMP Task", configMINIMAL_STACK_SIZE, NULL, 1, &xTaskHandles[1] );
+
+    vTaskStartScheduler();
+}
+
+/*
+The kernel will be configured as follows:
+    #define configNUMBER_OF_CORES                           (N > 1)
+    #define configUSE_CORE_AFFINITY                         1
+    #define configUSE_TASK_PREEMPTION_DISABLE               1
```

Comment:
We can skip this section since the test folder matches the FreeRTOSConfig.h.

Reply:
I noticed every other test_* function in covg_multiple_priorities_no_timeslice_utest.c have this section even though it is the same, so I haven't removed it yet. Waiting for confirmation.

Reply:
Removed the config section from comments in all new tests.